tests/server/test_environement_metrics.py

```diff
@@ -279,3 +282,86 @@ async def test_flush_metrics_mix(env_metrics_service):
     result_timer = await data.EnvironmentMetricsTimer.get_list()
     assert len(result_gauge) == 3
     assert len(result_timer) == 9
+
+
+async def test_resource_count_metric(clienthelper, environment, client, agent):
+    metrics_service = EnvironmentMetricsService()
+    version = str(await clienthelper.get_version())
+    resources = [
+        {
+            "key": "key1",
+            "value": "value1",
+            "id": "test::Resource[agent1,key=key1],v=" + version,
+            "send_event": False,
+            "purged": False,
+            "requires": [],
+        },
+        {
+            "key": "key2",
+            "value": "value2",
+            "id": "test::Resource[agent1,key=key2],v=" + version,
+            "send_event": False,
+            "requires": [],
+            "purged": False,
+        },
+        {
+            "key": "key3",
+            "value": "value3",
+            "id": "test::Resource[agent1,key=key3],v=" + version,
+            "send_event": False,
+            "requires": [],
+            "purged": True,
+        },
+    ]
+
+    result = await client.put_version(
```

Comment:
I would also add some resources to a different environment to cover the bug that Wouter noticed.